File: java-pi-pantry/src/test/java/com/langrsoft/util/JSonUtilParse.java
package com.langrsoft.util;

import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;
import static org.hamcrest.Matchers.is;

public class JSonUtilParse {
    static class X {
        public int y;
    }

    @Test
    public void createsInstanceOfClassFromJson() {
        X x = JsonUtil.parse("{\"y\": 42}", X.class);

        assertThat(x.y, is(equalTo(42)));
    }

    @Test(expected=JsonParseException.class)
    public void throwsRuntimeExceptionOnJsonParseException() {
        JsonUtil.parse("?", X.class);
    }
}

File: csharp-pi-pantry/csharp-pi-pantry-test/UtilTests/JSonUtilParse.cs
using Newtonsoft.Json;
using NUnit.Framework;
using Pipantry.Domain;
using Pipantry.Util;

namespace Test.Pipantry.Util
{
    public class JSonUtilParse
    {
        public class X
        {
            public int Y { get; set; }
        }

        [Test]
        public void CreatesInstanceOfClassFromJson()
        {
            X x = JsonUtil.Parse<X>("{\"Y\": 42}");

            Assert.That(x.Y, Is.EqualTo(42));
        }

        [Test]
        public void ThrowsRuntimeExceptionOnJsonParseException()
        {
            Assert.Throws<JsonParseException>(() => JsonUtil.Parse<X>("{\"Y\": \"not an int\"}"));
        }
    }
}
